descabi-morello/descabi-morello.rst

```diff
@@ -0,0 +1,769 @@
+    |                        | Data capability         | 16         | 16                        |                                        |
+    +------------------------+-------------------------+------------+---------------------------+----------------------------------------+
+
+
+Types Varying by Data Model and Procedure Calling Standard
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. _C/C++ type variants by data model and PCS:
+
+.. class:: aapcs64-morello-desc-c-cpp-type-variants
+
+.. table:: C/C++ type variants by data model and PCS
+
+    +-----------------------------+-------------------------------------+------------------------------+
+    | C/C++ Type                  | Machine Type                        | Notes                        |
+    +=============================+=====================================+==============================+
+    | ``T *``                     | Data Capability                     | Any data type ``T``.         |
+    +-----------------------------+-------------------------------------+------------------------------+
+    | ``T (*F)()``                | Function Capability                 | Any function type ``F``.     |
+    +-----------------------------+-------------------------------------+------------------------------+
+    | ``T&``                      | Data Capability                     | C++ reference.               |
+    +-----------------------------+-------------------------------------+------------------------------+
+    | ``T * __capability``        | Function Capability                 | Any data type ``T``.         |
+    +-----------------------------+-------------------------------------+------------------------------+
+    | ``T (* __capability F)()``  | Function Capability                 | Any function type ``F``.     |
+    +-----------------------------+-------------------------------------+------------------------------+
+    | ``T& __capability``         | Data Capability                     | C++ reference.               |
+    +-----------------------------+-------------------------------------+------------------------------+
+    | ``&&T``                     | Code Capability                     | Any label T.                 |
+    +-----------------------------+-------------------------------------+------------------------------+
+
+Morello Descriptor ABI ELF Extensions
+=====================================
+
+
+Private data sections
+---------------------
+
+The following sections are now part of a ``PT_MORELLO_DESC`` segment with the ``p_type`` value 0x70001000:
+
+- ``.desc.data.rel.ro``
+
+- ``.got``
+
+- ``.data``
+
+- ``.bss``
+
+
+The value of the Capability Private Data Addressing Register (c28) points to the start of the ``PT_MORELLO_DESC``
+segment, is read-only and can be used to access the ``.desc.data.rel.ro`` and ``.got`` sections.
```

Comment:
Does it need to cover `.desc.data.rel.ro` if all descriptors are accessed indirectly via the GOT?

Reply:
Currently .desc.data.rel.ro contains capabilities that are needed for objects which don't have an associated GOT slot (this is inherited from purecap). We could use a different section for these capabilities and not have to cover .desc.data.rel.ro. This would minimize privilege, but I wonder if it adds much security-wise.